### PR TITLE
Move tar commands into containers/storage

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@QiWang19 PTAL

**Comment by nalind** [comment]
What's the rationale for moving this instead of making it public?

**Comment by rhatdan** [comment]
The interfaces in buildah were fund (b *builder) and @QiWang19 copied and rewrote them for the podman-cp command, so I figured they were calling the chrootarchive util package so it made sense to move them there, and pull out the buidah specific calls.

They certainly are not buildah specific.

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by TomSweeneyRedHat** [comment]
Hmm, Red Hat CI isn't happy anywhere ATM.

File: vendor.conf [hunk 1]
```diff
@@ -6,7 +6,7 @@ github.com/containernetworking/cni v0.7.0-alpha1
 github.com/containers/image v1.3
 github.com/boltdb/bolt v1.3.1
 github.com/containers/libpod v1.0
-github.com/containers/storage v1.5
+github.com/containers/storage af01fe098c3eda83341f7c0f764956090efc5170 https://github.com/rhatdan/storage
```
Comment (TomSweeneyRedHat) [inline vendor.conf]: I'm assuming this will move to containers/storage before being merged?

Comment (rhatdan) [reply vendor.conf]: yup

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
bot, retest this please

File: util.go [hunk 2]
```diff
@@ -88,89 +85,31 @@ func convertRuntimeIDMaps(UIDMap, GIDMap []rspec.LinuxIDMapping) ([]idtools.IDMa
...
-			}
-			hashWorker.Wait()
-			if err == nil {
-				err = errors.Wrapf(hashError, "error calculating digest of data for %q while copying", dest)
-			}
-			return err
-		}
-	}
-	return archiver.CopyFileWithTar
+	return chrootarchive.CopyFileWithTar(chownOpts, hasher, convertedUIDMap, convertedGIDMap)
```
Comment (QiWang19) [inline util.go]: should this be `chrootarchive.CopyFileWithTarAndChown` ?

File: util.go [hunk 3]
```diff
@@ -88,89 +85,31 @@ func convertRuntimeIDMaps(UIDMap, GIDMap []rspec.LinuxIDMapping) ([]idtools.IDMa
...
-	untarMappings := idtools.NewIDMappingsFromMaps(convertedUIDMap, convertedGIDMap)
-	archiver := chrootarchive.NewArchiverWithChown(nil, chownOpts, untarMappings)
-	if hasher != nil {
-		originalUntar := archiver.Untar
-		archiver.Untar = func(tarArchive io.Reader, dest string, options *archive.TarOptions) error {
-			return originalUntar(io.TeeReader(tarArchive, hasher), dest, options)
-		}
-	}
-	return archiver.CopyWithTar
+	return chrootarchive.CopyWithTar(chownOpts, hasher, convertedUIDMap, convertedGIDMap)
```
Comment (QiWang19) [inline util.go]: also `chrootarchive.CopyWithTarAndChown` ?

File: util.go [hunk 4]
```diff
@@ -88,89 +85,31 @@ func convertRuntimeIDMaps(UIDMap, GIDMap []rspec.LinuxIDMapping) ([]idtools.IDMa
...
-	untarMappings := idtools.NewIDMappingsFromMaps(convertedUIDMap, convertedGIDMap)
-	archiver := chrootarchive.NewArchiverWithChown(nil, chownOpts, untarMappings)
-	if hasher != nil {
-		originalUntar := archiver.Untar
-		archiver.Untar = func(tarArchive io.Reader, dest string, options *archive.TarOptions) error {
-			return originalUntar(io.TeeReader(tarArchive, hasher), dest, options)
-		}
-	}
-	return archiver.UntarPath
+	return chrootarchive.UntarPath(chownOpts, hasher, convertedUIDMap, convertedGIDMap)
```
Comment (QiWang19) [inline util.go]: `chrootarchive.UntarPathAndChown`?

**Comment by rhatdan** [comment]
@rh-atomic-bot r=TomSweeneyRedHat 
One is supposed to be using a chroot while the other does not.

**Pull request closed without merging**